plugin-mainmenu/xdgcachedmenu.cpp

```diff
@@ -95,15 +96,32 @@ void XdgCachedMenu::addMenuItems(QMenu* menu, MenuCacheDir* dir)
     }
     else
     {
+      bool appVisible = type == MENU_CACHE_TYPE_APP
+          && menu_cache_app_get_is_visible(MENU_CACHE_APP(item),
+                                           menu_cache_desktop_env_flag_);
+      bool dirVisible = type == MENU_CACHE_TYPE_DIR
+          && menu_cache_dir_is_visible(MENU_CACHE_DIR(item));
+
+      if(!appVisible && !dirVisible)
+        continue;
+
       XdgCachedMenuAction* action = new XdgCachedMenuAction(item, menu);
       menu->addAction(action);
+
       if(type == MENU_CACHE_TYPE_APP)
         connect(action, SIGNAL(triggered(bool)), SLOT(onItemTrigerred()));
       else if(type == MENU_CACHE_TYPE_DIR)
       {
         XdgCachedMenu* submenu = new XdgCachedMenu(menu);
         action->setMenu(submenu);
-        addMenuItems(submenu, (MenuCacheDir*)item);
+        addMenuItems(submenu, MENU_CACHE_DIR(item));
+
+        if(submenu->actions().count() == 0)
```

Comment:
> > OK. The spec is pretty clear as well. I'll remove this code and we can discuss adding `show_empty="true"` in our menu file. But the "Other" category **is** shown, despite being empty, in my system.

> I would like to see your cache file, please. It might be some bug not
found yet in menu-cache after all. Thank you.

It may also be related to findings depicted here -> https://github.com/lxde/lxqt/issues/65#issuecomment-162634423